mysys/ma_dyncol.c

```diff
@@ -3865,17 +3865,18 @@ my_bool dynstr_append_json_quoted(DYNAMIC_STRING *str,
     }
     else
     {
+      if (lim < 2)
```

Comment:
so we moved this because there could be up to 2 more characters. We could be over allocating, but I'm not sure it matters.

Reply:
Over allocating should be fine, it's just a buffer.